R/00_eigs.R

```diff
@@ -237,30 +242,38 @@ eigs.dgeMatrix <- function(A, k, which = "LM", sigma = NULL, opts = list(), ...)
 ##' @export
 eigs.dgCMatrix <- function(A, k, which = "LM", sigma = NULL, opts = list(), ...)
 {
-    if(isSymmetric(A) &
-           which %in% c("LM", "SM", "LR", "SR") &
-           (is.null(sigma) || Im(sigma) == 0))
-    {
-        if(which == "LR")  which = "LA"
-        if(which == "SR")  which = "SA"
-        eigs_real_sym(A, nrow(A), k, which, sigma, opts, mattype = "sym_dgCMatrix",
+    if (is(A, 'symmetricMatrix') || (
+        isSymmetric(A) &&
+        which %in% c("LM", "SM", "LR", "SR") &&
+        (is.null(sigma) || Im(sigma) == 0)
+    )) {
+        if (which == "LR") which = "LA"
+        if (which == "SR") which = "SA"
+        eigs_real_sym(A, nrow(A), k, which, sigma, opts, mattype = "dsCMatrix",
                       extra_args = list(use_lower = TRUE))
```

Comment:
I think this should be split: `extra_args` should only be passed when calling this on a `dgCMatrix`, or we should forget about `extra_args` here and convert symmetric any `dgCMatrix` to a `dsCMatrix`.